### PR TITLE
T7742 alert component

### PR DESCRIPTION
**Pull request by vittoriaThinkst**
## Proposed changes

!! This pull request adds changes to many files, most of which are out of the scope of this ticket ( See at the end for details )
!! Relevant to the Notification components is:
- BaseMessageBox

Add Message Box component
Add Unit tests

The initial idea was to have just a message box for alerts, but we might have a couple of places where we show different typologies of messages.

Screenshot 2024-03-21 at 21 26 49
Screenshot 2024-03-21 at 21 26 57

Out of the ticket's scope:
- Add global export for Base component: unplugin-vue-components/vite automatically exports the Base components and makes them available to the global scope
- other fixes for Types, add coverage info on testing

### CODE REVIEW

**Review by CybergalxU — CHANGES_REQUESTED** [review]
File was not deleted. Access to the file was restored by an external user aka my wife. She hates me lol

File: frontend_vue/package.json [hunk 1]
```diff
@@ -53,6 +53,7 @@
     "sass": "^1.71.1",
     "tailwindcss": "^3.4.1",
     "typescript": "~5.3.0",
+    "unplugin-vue-components": "^0.26.0",
```
Comment (mclmax) [inline frontend_vue/package.json:56]: whats the benefit of this besides not having to import things?

File: frontend_vue/src/components/base/types.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+export interface BaseStateVariantType {
+  danger: unknown;
+  warning: unknown;
+  info: unknown;
```
Comment (mclmax) [inline frontend_vue/src/components/base/types.ts:4]: what's the unknown type? *? any reason not to make it string?

**Review by mclmax — APPROVED** [review]

File: frontend_vue/package.json (hunk 1 above)
Comment (vittoriaThinkst) [reply frontend_vue/package.json:56]: We don't have to export them manually: to make typescript work correctly with the Base components ( meaning: perform a type checking on props for instance ) we had to export them manually, by including them in the globalcomponents.d.ts file

File: frontend_vue/src/components/base/types.ts (hunk 2 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/components/base/types.ts:4]: oh! Nope absolutely no reason, it's a leftover!
I push a change and make them strings 🙏
thanks for catching it

**Pull request merged**